# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/service-workers/service-worker/resources/local-url-inherit-controller-frame.html
# Commit: 244df402d6eb
# Full Hash: 244df402d6ebc15d6845c4144245511021408835
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2024-10-24 09:44:34
# Regressor Bug: 1131324
# File Overlap Count: 1
# Description:
#   Bug 1131324 - Expose ServiceWorkerContainer on WorkerNavigator. r=dom-worker-reviewers,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D213725
# ==============================================================================

diff -r 57f70ba5f27e -r 244df402d6eb testing/web-platform/tests/service-workers/service-worker/resources/local-url-inherit-controller-frame.html
--- a/testing/web-platform/tests/service-workers/service-worker/resources/local-url-inherit-controller-frame.html	Thu Oct 24 03:02:39 2024 +0000
+++ b/testing/web-platform/tests/service-workers/service-worker/resources/local-url-inherit-controller-frame.html	Thu Oct 24 03:02:40 2024 +0000
@@ -140,7 +140,8 @@
   let w = new SharedWorker(url);
   return new Promise((resolve, reject) => {
     w.port.onmessage = m => {
-      if (m.data.includes("Error")) {
+      // (null is a valid value when the SharedWorker is not controlled)
+      if (m.data?.includes("Error")) {
         reject(m.data);
         return;
       }
